src/ACEtk/interpretation/ContinuousEnergyNeutron/fissionMultiplicity/Tabulated/src/ctor.hpp

```diff
@@ -5,23 +5,31 @@ Tabulated( ENDFInterpolationParameters&& interp,
   energies_( energies ),
   nubar_( nubar )
 {
-  assert( energies_.size() == nubar.size() );
+  if( energies_.size() != nubar.size() ){
+    njoy::Log::error( "incident energies and nubar must be the same size" );
+    njoy::Log::info( "energies size: {}, nubar size: {}", 
+                     energies_.size(), nubar_.size() );
+    throw std::exception();
+  }
 
-#if ACETK_CHECK
-  // sorted energies
-  // sorted NBT
-  // positive nubar
-  // INT must be in [1,8)
-#endif
+  if( not details::verify::sorted_positive( energies_ ) ){
+    njoy::Log::info( "incident energies must be sorted" );
+    throw std::exception();
+  }
+
+  if( not details::verify::positive( nubar ) ){
+    njoy::Log::info( "nubar values must be positive" );
+    throw std::exception();
+  }
 }
 
 template< typename Range >
```

Comment:
This really needs to be constrained

Reply:
Constrained by what?

Reply:
There's  no check that the parameter type meets an expected interface